bindings/java/src/main/native/context.cc

```diff
@@ -52,10 +92,30 @@ Java_org_openucx_jucx_ucp_UcpContext_createContextNative(JNIEnv *env, jclass cls
     ucp_params.request_size = sizeof(struct jucx_context);
     ucp_params.request_init = jucx_request_init;
 
-    ucs_status_t status = ucp_init(&ucp_params, NULL, &ucp_context);
+    ucp_config_t *config = NULL;
+    ucs_status_t status;
+
+    field = env->GetFieldID(jucx_param_class, "config", "Ljava/util/Map;");
+    jobject config_map = env->GetObjectField(jucx_ctx_params, field);
+
+    if (config_map != NULL) {
+        status = ucp_config_read(NULL, NULL, &config);
+        if (status != UCS_OK) {
+            JNU_ThrowExceptionByStatus(env, status);
+        }
+
+        iterate_java_map(env, config, &config_map);
+    }
+
+    status = ucp_init(&ucp_params, config, &ucp_context);
     if (status != UCS_OK) {
         JNU_ThrowExceptionByStatus(env, status);
     }
+
+    if (config != NULL) {
+        ucp_config_release(config);
```

Comment:
need to release config if ucp_init fails

Reply:
It'll come here. `JNU_ThrowExceptionByStatus` doesn't return from the method, just throws java exception and continues execution.